ui/packages/atlasmap/src/Views/MappingTableView.tsx

```diff
@@ -46,25 +48,98 @@ const styles = StyleSheet.create({
 });
 
 export interface IMappingTableProps {
-  mappings: IAtlasmapMapping[];
+  properties?: IAtlasmapDocument | null;
+  constants?: IAtlasmapDocument | null;
+  sources: Array<IAtlasmapDocument>;
+  mappings: Array<IAtlasmapMapping>;
+  targets: Array<IAtlasmapDocument>;
   onSelectMapping: (mapping: IAtlasmapMapping) => void;
+  shouldShowMappingPreviewForField: (field: IAtlasmapField) => boolean;
+  onFieldPreviewChange: (field: IAtlasmapField, value: string) => void;
 }
 
 export const MappingTableView: FunctionComponent<IMappingTableProps> = ({
+  properties,
+  constants,
+  sources,
   mappings,
+  targets,
   onSelectMapping,
+  shouldShowMappingPreviewForField,
+  onFieldPreviewChange,
 }) => {
+  const find = (
+    mappingField: IAtlasmapField,
+    fields: AtlasmapFields,
+  ): IAtlasmapField | undefined => {
+    let field = undefined;
+    fields.every((f) => {
+      if (f.id === mappingField.id) {
+        field = f;
+        return false;
+      }
+      if ((f as IAtlasmapGroup).fields) {
+        field = find(mappingField, (f as IAtlasmapGroup).fields);
+        return field === undefined;
+      }
+      return true;
+    });
+    return field;
+  };
+
+  const renderPreview = (mappingField: IAtlasmapField) => {
+    let field = properties?.fields.find((f) => f.id === mappingField.id);
+    if (field === undefined) {
+      field = constants?.fields.find((f) => f.id === mappingField.id);
+    }
+    if (field === undefined) {
+      sources.every((source) => {
+        field = find(mappingField, source.fields);
+        return field === undefined;
+      });
+    }
+    if (field === undefined) {
+      targets.every((target) => {
+        field = find(mappingField, target.fields);
+        return field === undefined;
+      });
+    }
+    const f = field as IAtlasmapField;
+    return (
+      f &&
+      shouldShowMappingPreviewForField(f) && (
+        <DocumentFieldPreview
+          id={f.id}
+          value={f.previewValue}
+          onChange={(value) => onFieldPreviewChange(f, value)}
+        />
+      )
+    );
+  };
+
   const rows =
     mappings.length === 0
       ? emptyContent
       : mappings.map((mapping) => {
-          const sources = mapping.sourceFields.map((source) => {
-            const { name } = source as IAtlasmapField;
-            return <div key={name}>{name}</div>;
+          const sources = mapping.sourceFields.map((source, index) => {
```

Comment:
I'm not sure I'm following, why are you searching for the field again when the field is already available?

Reply:
I assume you mean the search inside renderPreview that starts with the field from the mapping. The reason is the mapping fields are missing critical information, such as the preview value and field.mappings is always empty, which causes the shouldShowMappingPreviewForField function in Atlasmap.tsx to always return false.

Reply:
Well, we don't really need `shouldShowMappingPreviewForField` in this case, a `shouldShowMappingPreview: (mapping: IAtlasmapMapping) => boolean` should be enough since then all the fields of a selected mapping that should show a preview should show a preview. 

And for the missing data I'd work on [`fromMappedFieldToIMappingField`](https://github.com/atlasmap/atlasmap/blob/master/ui/packages/atlasmap/src/Atlasmap/utils/ui.ts#L133-L178) to sync how a field is generated, by returning an enhanced version of [`fromFieldToIFieldsNode`](https://github.com/atlasmap/atlasmap/blob/master/ui/packages/atlasmap/src/Atlasmap/utils/ui.ts#L69-L96).

Reply:
I did it this way to reuse existing code, but I can look at a simpler version that just checks whether the mapping is selected.